File: topic_test/src/my_test_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

int main(int argc, char** argv)
{
    ros::init(argc, argv, "my_test_publisher");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Int64>("test_topic", 100);

    ros::Rate loop_rate(4);

    std_msgs::Int64 msg;
    msg.data = 0;

    while (ros::ok())
    {
        pub.publish(msg);
        loop_rate.sleep();
        msg.data++;
        if (msg.data == 101)
        {
            msg.data = 0;
        }
    }

    return 0;
}

File: topic_test/scripts/my_test_publisher.py
#! /usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from std_msgs.msg import Int64

def my_counter():
    rospy.init_node("my_test_publisher")
    pub = rospy.Publisher("my_count", Int64, queue_size=100)

    loop_rate = rospy.Rate(4)

    msg = Int64()
    msg.data = 0

    while not rospy.is_shutdown():
        pub.publish(msg)
        loop_rate.sleep()
        msg.data += 1
        if (msg.data == 101):
            msg.data = 0

if __name__ == "__main__":
    try :
        my_counter()
    except rospy.ROSInterruptException:
        pass
